File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        set_nums = set(nums)
        longest = 0

        # Calculate the start of sequence
        for n in nums:
            if n - 1 not in nums:
                length = 0
                while n + length in nums:
                    length += 1
                longest = max(longest, length)

        return longest

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxProfit = 0;
        int l = 0;
        int r = 0;

        while(r < prices.length){
            if(prices[r] < prices[l]){
                l = r;
            } else {
                maxProfit = Math.max(maxProfit, prices[r] - prices[l]);
            }
            r++;
        }
        return maxProfit;
    }
}
